server.js

```diff
@@ -45,157 +53,164 @@ function main () {
-                async.waterfall([
-                    async.apply(getGameInfo, ApiKey, toSend, championsMap, region, summonerName, version),
-                ], function (err, ApiKey, toSend, championsMap, region, summonerName, version) {
-                    if (!err) {
-                        socket.emit("info sent", toSend);
-                        called = false;
-                    }
-                });
+                var a = sendInfo();
+
+              function sendInfo() {
+                Promise.all([
+                    getGameInfo(region),
+                    getPlayerInfo(region, summonerName)
+                  ])
+                  .then(values => {
+                    var send = mergeInfo(values[0], values[1]);
+                    socket.emit("info sent", send);
+                    called = false;
+                  });
+
+                function mergeInfo(versionChampions, playerInfo){
+                  let toSend = {};
+
+                  let championsMap = versionChampions[1];
+                  let champMasteries = playerInfo[1];
+                  toSend.infos = playerInfo[0];
+
+                  for (let i = 0; i < champMasteries.length; ++i) {
+                    let champId = champMasteries[i].championId;
+                    let urlSquare = "http://ddragon.leagueoflegends.com/cdn/" + versionChampions[0] + "/img/champion/" + championsMap[champId][0] + ".png";
+                    champMasteries[i].name = championsMap[champId][1];
+                    champMasteries[i].riotName = championsMap[champId][0];
+                    champMasteries[i].urlImage = urlSquare;
+
+                    champId = championsMap[champId];
+                  }
+                  toSend.champMasteries = champMasteries;
+                  return toSend;
+                }
+              }
             }
         });
     });
 }
 
-function getGameInfo(ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    async.waterfall([
-        async.apply(getApiKey, pathToYourApiKey, ApiKey, toSend, championsMap, region, summonerName, version),
-        getVersion,
-        getChampionsJson,
-        getPlayerInfo
-    ], function (err, ApiKey, toSend, championsMap, region, summonerName, version) {
-        callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-    });
-}
+function getGameInfo(region) {
+  function logError(e) {
+    console.error("error is:" + e);
+    throw (e);
+  }
+  var a = getVersion(region);
+  var b = a.then(function (version) {
+    let req = "http://ddragon.leagueoflegends.com/cdn/" + version + "/data/en_US/champion.json " ;
 
-function getApiKey(file, ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    ApiKey = require(file);
-    callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-}
+    return new Promise(function(resolve, reject) {
+      request(req, function (error, response, body) {
```

Comment:
You are using request-promise so you don't need to put a callback in second argument, only `.then` to use it as a promise.